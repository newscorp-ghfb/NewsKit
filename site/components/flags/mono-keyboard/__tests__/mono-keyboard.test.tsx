import React from 'react';
import {
  renderToFragmentWithTheme,
  renderWithTheme,
} from '../../../../utils/test-utils';
import {MonoKeyboard} from '..';

function mockComp(name: string) {
  return ({children, ...props}: any) => (
    <span data-testid={name} data-props={JSON.stringify(props, null, 2)}>
      {children}
    </span>
  );
}

jest.mock('../../mono', () => ({
  ...(jest.requireActual('../../mono') as any),
  Mono: mockComp('Mono'),
}));

describe('MonoKeyboard', () => {
  describe('with single key', () => {
    it('should render with no icon for unrecognised key', () => {
      const fragment = renderToFragmentWithTheme(MonoKeyboard, {
        children: 'Ctrl',
      });
      expect(fragment).toMatchSnapshot();
    });

    [
      ['Rtn', 'IconOutlinedKeyboardReturn'],
      ['Backspace', 'IconOutlinedKeyboardBackspace'],
      ['Down', 'IconOutlinedKeyboardArrowDown'],
      ['Left', 'IconOutlinedKeyboardArrowLeft'],
      ['Right', 'IconOutlinedKeyboardArrowRight'],
      ['Space', 'IconOutlinedSpaceBar'],
      ['Tab', 'IconOutlinedKeyboardTab'],
      ['Up', 'IconOutlinedKeyboardArrowUp'],
    ].forEach(([key, icon]) => {
      it(`should render the ${icon} for the ${key} key`, () => {
        const dom = renderWithTheme(MonoKeyboard, {
          children: key,
        });

        expect(dom.asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('with multiple keys', () => {
    it('should render as expected', () => {
      const fragment = renderToFragmentWithTheme(MonoKeyboard, {
        children: ['Ctrl', 'Rtn', 'Left'],
      });
      expect(fragment).toMatchSnapshot();
    });
  });
});
