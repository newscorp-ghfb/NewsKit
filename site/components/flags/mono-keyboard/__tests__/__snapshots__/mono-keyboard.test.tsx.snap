// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonoKeyboard with multiple keys should render as expected 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2e3f54;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-3.emotion-3 {
  width: 16px;
  height: 16px;
}

.emotion-6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-6 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-6 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-6.emotion-6 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-6.emotion-6 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-6.emotion-6 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        Ctrl
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-3 emotion-4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </div>
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-6 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 7v4H5.83l3.58-3.59L8 6l-6 6 6 6 1.41-1.41L5.83 13H21V7h-2z"
          />
        </svg>
        Rtn
      </span>
    </div>
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-3 emotion-4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </div>
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-6 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
        </svg>
        Left
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardArrowDown for the Down key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
        Down
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardArrowLeft for the Left key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
        </svg>
        Left
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardArrowRight for the Right key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
        Right
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardArrowUp for the Up key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
          />
        </svg>
        Up
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardBackspace for the Backspace key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21v-2z"
          />
        </svg>
        Backspace
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardReturn for the Rtn key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 7v4H5.83l3.58-3.59L8 6l-6 6 6 6 1.41-1.41L5.83 13H21V7h-2z"
          />
        </svg>
        Rtn
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedKeyboardTab for the Tab key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11.59 7.41L15.17 11H1v2h14.17l-3.59 3.59L13 18l6-6-6-6-1.41 1.41zM20 6v12h2V6h-2z"
          />
        </svg>
        Tab
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render the IconOutlinedSpaceBar for the Space key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #3358cc;
  fill: #3358cc;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2.emotion-2 {
  width: 16px;
  height: 16px;
  color: #3358cc;
  fill: #3358cc;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2.emotion-2 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        <svg
          aria-hidden="true"
          class="emotion-2 emotion-3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M18 9v4H6V9H4v6h16V9h-2z"
          />
        </svg>
        Space
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MonoKeyboard with single key should render with no icon for unrecognised key 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <span
        data-props="{}"
        data-testid="Mono"
      >
        Ctrl
      </span>
    </div>
  </div>
</DocumentFragment>
`;
